src/mount.c

```diff
@@ -32,38 +34,46 @@ pu_create_mount_point(const gchar *name)
 
 gboolean
 pu_mount(const gchar *source,
-         const gchar *mount_point)
+         const gchar *mount_point,
+         GError **error)
 {
+    g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
     g_autofree gchar *cmd = g_strdup_printf("mount %s %s", source, mount_point);
 
-    if (!g_spawn_command_line_sync(cmd, NULL, NULL, NULL, NULL)) {
-        g_critical("Failed mounting '%s' to '%s'!", source, mount_point);
+    if (!g_spawn_command_line_sync(cmd, NULL, NULL, NULL, error)) {
+        g_prefix_error(error, "Failed mounting: ");
         return FALSE;
     }
 
     return TRUE;
 }
 
 gboolean
-pu_umount(const gchar *mount_point)
+pu_umount(const gchar *mount_point,
+          GError **error)
 {
+    g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
     g_autofree gchar *cmd = g_strdup_printf("umount %s", mount_point);
 
-    if (!g_spawn_command_line_sync(cmd, NULL, NULL, NULL, NULL)) {
-        g_critical("Failed unmounting '%s'!", mount_point);
+    if (!g_spawn_command_line_sync(cmd, NULL, NULL, NULL, error)) {
+        g_prefix_error(error, "Failed unmounting: ");
         return FALSE;
     }
 
     return TRUE;
 }
 
 gboolean
-pu_device_mounted(const gchar *device)
+pu_device_mounted(const gchar *device,
+                  GError **error)
```

Comment:
For this function we cannot use GError this way. Normally, functions with GError return `FALSE` if an error occured and `TRUE` on success. `pu_device_mounted()` however returns `FALSE` if the specified device is not mounted, but this is a valid case without error. I'd suggest leaving GError out of this function.

Maybe we should also rename it to something more meaningful, e.g.: `pu_device_is_mounted()`